### PR TITLE
refactor: fixing test and other small changes

### PR DESCRIPTION
**Pull request by asbiin**
- replacing Eloquent's `Model::where(...)->first()` with `Model::firstWhere(...)`
- add a `group_must_belong_to_vault` validate rule
- refactor validation rules in `BaseService` to ensure the order and dependencies between the rules

### CODE REVIEW

File: app/Domains/Contact/ManageContact/Web/Controllers/ContactLabelController.php [hunk 1]
```diff
@@ -18,7 +18,7 @@ public function index(Request $request, string $vaultId, int $labelId)
     {
         $vault = Vault::findOrFail($vaultId);
 
-        $contacts = Label::where('id', $labelId)->first()
+        $contacts = Label::find($labelId)
```
Comment (djaiss) [inline app/Domains/Contact/ManageContact/Web/Controllers/ContactLabelController.php:21]: Why `find` and not `first`?

File: app/Services/BaseService.php [hunk 2]
```diff
@@ -24,7 +26,45 @@ abstract class BaseService
...
+     * The contact object.
+     */
+    public ?Group $group = null;
+
+    /**
+     * Dependencies between permissions.
+     *
+     * @var array<string,array<string>>
+     */
+    private static array $dependencies = [
```
Comment (djaiss) [inline app/Services/BaseService.php:41]: Would `permissionDependencies` be a better name for this? `dependencies` alone seems a bit off.

File: app/Services/BaseService.php [hunk 3]
```diff
@@ -57,35 +97,60 @@ public function validateRules(array $data): bool
...
-            $this->validateAuthorIsAccountAdministrator();
-        }
+        $permissions = collect($this->permissions());
 
-        if (in_array('vault_must_belong_to_account', $this->permissions())) {
-            $this->validateVaultExists($data);
-        }
+        foreach (self::$dependencies as $key => $value) {
+            if ($permissions->contains($key)) {
+                collect($value)->each(function ($v) use ($permissions, $key) {
```
Comment (djaiss) [inline app/Services/BaseService.php:104]: Why `v` as a name?

**Review by djaiss — COMMENTED** [review]
A few comments. Otherwise, it's great.

File: app/Domains/Contact/ManageContact/Web/Controllers/ContactLabelController.php (hunk 1 above)
Comment (asbiin) [reply app/Domains/Contact/ManageContact/Web/Controllers/ContactLabelController.php:21]: `find(xx)` is actually the same as `where('id', xxx)->first()`

File: app/Services/BaseService.php (hunk 2 above)
Comment (asbiin) [reply app/Services/BaseService.php]: done, thanks for the suggestion

File: app/Services/BaseService.php (hunk 3 above)
Comment (asbiin) [reply app/Services/BaseService.php]: Correct, I've renamed it

**Pull request merged**